rocky/rocky/scheduler.py

```diff
@@ -238,9 +255,62 @@ def push_task(self, queue_name: str, prioritized_item: QueuePrioritizedItem) ->
                 raise SchedulerError()
 
     def health(self) -> ServiceHealth:
-        health_endpoint = self.session.get(f"{self._base_uri}/health")
-        health_endpoint.raise_for_status()
-        return ServiceHealth.parse_raw(health_endpoint.content)
+        try:
+            health_endpoint = self.session.get(f"{self._base_uri}/health")
+            health_endpoint.raise_for_status()
+            return ServiceHealth.parse_raw(health_endpoint.content)
+        except ConnectionError:
+            raise SchedulerError()
+
+
+def get_scheduler_client():
+    try:
+        client = SchedulerClient(settings.SCHEDULER_API)
+        client.health()
+        return client
+    except ConnectionError:
+        raise SchedulerError()
+
+
+def schedule_task(request: HttpRequest, organization_code: str, task: QueuePrioritizedItem) -> None:
+    plugin_name = ""
+    input_ooi = ""
+    if task.data.type == "boefje":
+        plugin_name = task.data.boefje.name
+        input_ooi = task.data.input_ooi
+    if task.data.type == "normalizer":
+        plugin_name = task.data.normalizer.id  # name not set yet, is None for name
+        input_ooi = task.data.raw_data.boefje_meta.input_ooi
+    try:
+        get_scheduler_client().push_task(f"{task.data.type}-{organization_code}", task)
+    except (BadRequestError, TooManyRequestsError, ConflictError) as task_error:
+        error_message = task_error.message + _(" Scheduling {} {} with input object {} failed.").format(
+            task.data.type.title(), plugin_name, input_ooi
+        )
+        messages.error(request, error_message)
+    except SchedulerError as error:
+        messages.error(request, error.message)
+    else:
+        messages.success(
+            request,
+            _(
+                "Task '{} {} with input object {}' is scheduled and will soon be started in the background. "
+                "Results will be added to the object list when they are in. "
+                "It may take some time, a refresh of the page may be needed to show the results."
+            ).format(task.data.type.title(), plugin_name, input_ooi),
+        )
+
+
+def get_list_of_tasks_lazy(request: HttpRequest, **params) -> LazyTaskList | None:
```

Comment:
The return type doesn't make sense, either `None` should be a list, or the returned list should be a `None` instead